File: old_trunk/cost/iona/orbacus/cpp/poa/TestServer_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <Test_skel.h>

#include <TestServer_impl.h>

using namespace CORBA;
using namespace PortableServer;

TestServer_impl::TestServer_impl(ORB_ptr orb, const TestInfoSeq& info)
    : orb_(ORB::_duplicate(orb)), info_(info)
{
}

TestServer_impl::~TestServer_impl()
{
}

TestInfoSeq*
TestServer_impl::get_info()
    throw(SystemException)
{
    return new TestInfoSeq(info_);
}

void
TestServer_impl::deactivate()
    throw(SystemException)
{
    if(!CORBA::is_nil(orb_))
    {
	orb_ -> shutdown(false);
	orb_ = CORBA::ORB::_nil();
    }
}

File: old_trunk/cost/iona/orbacus/java/poa/TestServer_impl.java
// **********************************************************************
//
// Copyright (c) 2000
// Object Oriented Concepts, Inc.
// Billerica, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package test.poa;

import org.omg.CORBA.*;

final class TestServer_impl extends TestServerPOA
{
    private ORB orb_;
    private TestInfo[] info_;

    TestServer_impl(ORB orb, TestInfo[] info)
    {
	orb_ =orb;
	info_ = info;
    }

    public TestInfo[]
    get_info()
    {
	return info_;
    }

    public void
    deactivate()
    {
	if(orb_ != null)
	{
	    orb_.shutdown(false);
	    orb_ = null;
	}
    }
}
